### DISCUSSION TITLE
Zodiac Flow Pro

### DISCUSSION THREAD

**Discussion by odgeza** [aqualinkd/AqualinkD#363] [General]
Hi All

New to all this, but by the looks of it, this system is PDA type system : Zodiac VS flopro
<img width="458" alt="msedge_0HqH0p0TeE" src="https://github.com/user-attachments/assets/5e21112d-a226-4c87-a540-569afc8338e6">

![ApplicationFrameHost_92RM4ekx39](https://github.com/user-attachments/assets/011c982b-9d8c-4e8a-92d2-246ba2b1a693)

I see a Zodiac mentioned here. It looks the same as Jandy VS Flowpro 

Someone mentioned the PDA here. https://github.com/sfeakes/AqualinkD/discussions/277

I'm getting a lot of "reads" on the USB adapter, tons of info on sudo -s eval 'stty raw -echo < /dev/ttyUSB0; cat -vte /dev/ttyUSB0'

green light crazy, but nothing from serial logger

```
admin@raspberrypi:~ $ dmesg | grep usb
[    0.089247] usbcore: registered new interface driver usbfs
[    0.089322] usbcore: registered new interface driver hub
[    0.089403] usbcore: registered new device driver usb
[    1.817765] usbcore: registered new device driver r8152-cfgselector
[    1.819787] usbcore: registered new interface driver r8152
[    1.821796] usbcore: registered new interface driver lan78xx
[    1.823662] usbcore: registered new interface driver smsc95xx
[    2.786901] dwc_otg 3f980000.usb: DWC OTG Controller
[    2.788428] dwc_otg 3f980000.usb: new USB bus registered, assigned bus number 1
[    2.789987] dwc_otg 3f980000.usb: irq 74, io mem 0x00000000
[    2.794942] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 6.06
[    2.798116] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    2.799802] usb usb1: Product: DWC OTG Controller
[    2.801473] usb usb1: Manufacturer: Linux 6.6.51+rpt-rpi-v8 dwc_otg_hcd
[    2.803177] usb usb1: SerialNumber: 3f980000.usb
[    2.810689] usbcore: registered new interface driver uas
[    2.812457] usbcore: registered new interface driver usb-storage
[    2.830966] usbcore: registered new interface driver usbhid
[    2.832648] usbhid: USB HID core driver
[    3.112358] usb 1-1: new high-speed USB device number 2 using dwc_otg
[    3.324824] usb 1-1: New USB device found, idVendor=0424, idProduct=2514, bcdDevice= b.b3
[    3.328207] usb 1-1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    3.628325] usb 1-1.1: new high-speed USB device number 3 using dwc_otg
[    3.729091] usb 1-1.1: New USB device found, idVendor=0424, idProduct=2514, bcdDevice= b.b3
[    3.736397] usb 1-1.1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    4.036383] usb 1-1.1.2: new full-speed USB device number 4 using dwc_otg
[    4.164090] usb 1-1.1.2: New USB device found, idVendor=0403, idProduct=6001, bcdDevice= 6.00
[    4.168093] usb 1-1.1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    4.170209] usb 1-1.1.2: Product: TTL232RG-VREG3V3
[    4.172290] usb 1-1.1.2: Manufacturer: FTDI
[    4.174302] usb 1-1.1.2: SerialNumber: FT54N0MY
[    4.440432] usb 1-1.1.1: new high-speed USB device number 5 using dwc_otg
[    4.541222] usb 1-1.1.1: New USB device found, idVendor=0424, idProduct=7800, bcdDevice= 3.00
[    4.545096] usb 1-1.1.1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    9.628183] usbcore: registered new interface driver usbserial_generic
[    9.632423] usbserial: USB Serial support registered for generic
[    9.723870] usbcore: registered new interface driver ftdi_sio
[    9.723980] usbserial: USB Serial support registered for FTDI USB Serial Device
[    9.740695] usb 1-1.1.2: Detected FT232R
[    9.760469] usb 1-1.1.2: FTDI USB Serial Device converter now attached to ttyUSB0
[   10.621774] usbcore: registered new interface driver brcmfmac
```
```

Please wait.admin@raspberrypi:~ $ sudo ~/software/AqualinkD/release/serial_logger-arm64 /dev/ttyUSB0
AqualinkD serial_logger V2.7
Notice:  RS Serial: Port /dev/ttyUSB0 low latency mode is set
Notice:  Serial Log:Logging serial information!


^CPlease wait.Notice:  Serial Log:Stopping!
^X^Z

```

I'm looking for any pointers, I dont know what to start doing next.

**Comment by sfeakes** [comment]
I think you might have data+ and data- (orange and yellow wires) round the wrong way.
I’m also not sure anyone has tried this on that specific Zodiac system.

but switch the wires first and see if that helps.   Also post output of tty dump command.

**Comment by odgeza** [comment]
Hi

I would have thought I would get no read date if it was the wrong way around? When I restart the pump, I also get a silence until it starts again.

tty output: (put a start date and end date for timing view)

```
admin@raspberrypi:~ $ date
Wed 20 Nov 08:17:52 SAST 2024
admin@raspberrypi:~ $ sudo -s eval 'stty raw -echo < /dev/ttyUSB0; cat -vte /dev/ttyUSB0'
yeM-7yM-iM-^?M-y^OM-{M-^?M-AuM-^?M-^?M-y^@^OM-yM-^?M-AM-^?^AM-_M-7u[M->M-7uM-yoM-_M-y^OM->M-7yM-iM-^?^A^@^OM-^?M-wM-yM-7uM-^?M-nM-#^@^OeM->M-^?M-AM-^?^AM-_M-7M-^?M-uM-yM-^?M-^?M-}M->^OM-^?M-^WM-^?M-^?M-AoIM-_M-7M->M-^?M-}W^@^OyM-yM-^?yM-^?M-^?M-y^@^OM-sUM-^?M-AM-^?M-sg^@^OeM->M-^?M-AyM-^?^AM-_M-7^OSM->M-7M-^?M-qM-^?M-_M-y^OyM-yM-^?yM-iM-^?^AM-_M-y^@M-7M-oQM-^?M-7M-^?M-mM->M-7^OM->M-yM-^?M-iM-^?^AM-_M-7^OM-mOM->M-^?M-^?M-m^A^@M-7eM->M-7yM-iM-^?M-_M-7^OuMM-^?M-AuM-QM-}^@M-7M-^?oM->M-yM-^?M-^?M-^?M-y^@M-7iM-^?M-}}M-_M-y^OM->M-7yM-^?^A^@^OM-iKM-^?M-7uM-^?M-i^B^@^OeM-^?M-AyM-^?M-^?M-y^@uSM->M-7u^A}M-y^OyM-^?M-AyM-iM-^?^AM-y^@^OM-^?M-yM-^?M-^?M-^?OM-_^@M-7eM->M-7M-^?M-AM-^?M-^?M-^?M-y^OM-}M-yM-^?M-7uM-^?M-}M->^@yM-^?M-7M-i^A^@^OM-{M-^?M-7uM-^?M-_M-y^OoM->M-7wM-^FM-_M-y^@^OM->M-7}M-_M-7eM->M-7M-i^AM-_M-7M-^?M->M-yM-^?M-AM-yoM-_M-7yeM->M-yM-^?M-iM-^?^A^@uYM->M-7M-^?M-n^@^OM->M-yM-^?M-iM-^?^A^@uWM->M-yM-^?M-uM->M-y^OyM-yM-^?M-AyM-iM-^?^AM-_M-7M-sUM-^?M-AM-sM-^?M-_M-y^OM->M-7yM-iM-^?^A^@uM-^?M-yM-^?M-AM-qM-eM->M-7M-^?M->M-^?M-^?oI^@^OM-_M-7}M-_M-y^OM-yM-^?M-AM-^?^AM-_M-7^OQM-7uM-^?M-oM-mM-y^@^OyM-yM-^?M-AM-iM-^?^AM-_M-7M-mM-yM-^?M-AM-mM-^?M-_M-y^OM->M-7M-iM-^?M-y^OMM->M-7uM-QM-}M-y^@^OM-yM-^?M-AyM-^?^A^@^OM-iKM-^?M-AM-^?^B^BM-y^OM->M-7M-i^AM-7M-}M->M-yM-7uM-}^BM-y^@^OM-^FM-^?M-AoM-_M-y^@}M-yM-^?M-}WM-_M-7eM-^?M-AM-^?^AM-_M-7M-^?aM->M-7uoM-^FM-y^@yM-^?M-AM-^?^AM-_M-7M-^?M->M-7uM-}M-'M->M-y^OM-yM-^?M-AM-iM-^?^A^@^OM-{M-yM-^?M-^?M-{M-^?M-_M-y^OM->M-7yM-^?M-^?M-y^@^OM-yM-yM-^?M-7uM-^?M-^FM-y^@^OyM->M-7yM-iM-^?^A^@^OM-wM->M-7uM-^_M-_M-y^OoM->M-yM-^?oM-^FM-_M-y^OM-yM-^?M-^?^@yM-^?M-AM-^?M-_^C
admin@raspberrypi:~ $ date
Wed 20 Nov 08:18:09 SAST 2024

```


Here is a Jandy side view
<img width="652" alt="msedge_KEZE0bIeGN" src="https://github.com/user-attachments/assets/ebe76630-91a6-46e3-9fc1-01cddba5fb80">

here is a Zodiac
<img width="644" alt="msedge_jmVTogveeW" src="https://github.com/user-attachments/assets/7a924408-5e31-4b91-9f14-e3557fad5e76">

I was also wonder, why is it the motor is not spoken to directly, (why on these PDA ones, do we speak to the PDA first).

**Reply by odgeza** [reply]
ok, so as you said, swap them around and it works.

Notice:  Serial Log:Logging serial information!
Please wait.|

Notice:  Serial Log:RS485 interface received 800 packets in 56 seconds ( 14.29 M                                             sg/Sec)
Notice:  Serial Log:Jandy Control Panel Model   :
Notice:  Serial Log:Jandy Control Panel Version :
Notice:  Serial Log:Jandy ID's found
Notice:  Serial Log:ID 0x78 is in use    <-- Jandy VSP ePump
Notice:  Serial Log:

Notice:  Serial Log:Suggested aqualinkd.conf values
Notice:  Serial Log:-------------------------
Notice:  Serial Log:read_RS485_ePump = yes
Notice:  Serial Log:-------------------------

I set that to true, but I am at a bit of a loss with the device ID. If I set it to 0x78

@sfeakes 

> Dec 08 12:51:26 raspberrypi systemd[1]: Starting aqualinkd.service - Aqualink RS daemon...
Dec 08 12:51:26 raspberrypi aqualinkd[837]: AqualinkD: Unknown config parameter 'log_file=/var/log/aqualinkd.log'
Dec 08 12:51:26 raspberrypi aqualinkd[837]: AqualinkD: Device ID 0x78 does not match PDA panel, please check config!
Dec 08 12:51:26 raspberrypi aqualinkd[837]: Error:   AqualinkD: Unknown config parameter 'log_file=/var/log/aqualinkd.log'
Dec 08 12:51:26 raspberrypi aqualinkd[837]: Error:   AqualinkD: Device ID 0x78 does not match PDA panel, please check config
Dec 08 12:51:26 raspberrypi systemd[1]: aqualinkd.service: Control process exited, code=exited, status=1/FAILURE
Dec 08 12:51:26 raspberrypi systemd[1]: aqualinkd.service: Failed with result 'exit-code'.
Dec 08 12:51:26 raspberrypi systemd[1]: Failed to start aqualinkd.service - Aqualink RS daemon.
admin@raspberrypi:~ $ sudo nano /etc/aqualinkd.conf

**Reply by sfeakes** [reply]
Serial Logger has not found your control panel.  So the pump setting on its own will not help much, as AqualinkD has to talk to the pump through the control panel, and not directly.
Can you let me know what control panel you have / version etc?

**Comment by odgeza** [comment]
Hi

It appears to be "Zodiac VSP 1.6HP 9AA84M V2.9" and "Zodiac VSP UI FW REV: C03"

![ApplicationFrameHost_1HF9RFQmPi](https://github.com/user-attachments/assets/9b27a631-bd4b-42e0-ad13-c81aca2482bd)


![image](https://github.com/user-attachments/assets/83a44099-9186-4a8f-b0b6-e24661547244)

![ApplicationFrameHost_179j97g1RN](https://github.com/user-attachments/assets/fe1d1d91-cecc-4f15-9368-19b7cd3b80de)

I tried using the control panel to set more settings, the serial logger never gave me more than it did previously, but  I also took the raw output while setting settings

`^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^FM-U^P^C^P^B^@^_E^@^FM-@PM-^L^P^C^P^BxD^@M-H^^M-4^P^C^P^B^@^_D^@M-H^^^@[^P^C^P^BxAM-K^P^C^P^B^@^AA^@T^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^GM-V^P^C^P^B^@^_E^@^GM-^@^S^P^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^HM-W^P^C^P^B^@^_E^@^H^K^@M-^I^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^IM-X^P^C^P^B^@^_E^@^I^@^@^?^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@$
M-Y^P^C^P^B^@^_E^@$
^Z^AM-^[^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^KM-Z^P^C^P^B^@^_E^@^KM-^?M-^?^?^P^C^P^BxD^@M-H^^M-4^P^C^P^B^@^_D^@M-H^^^@[^P^C^P^BxAM-K^P^C^P^B^@^AA^@T^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^A^FM-V^P^C^P^B^@^_E^A^FM-^?M-^?{^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^@M-O^P^C^P^B^@^_E^@^@M-<^^P^P^C^P^BxD^@M-h&M-\^P^C^P^B^@^_D^@M-h&^@M-^C^P^C^P^BxAM-K^P^C^P^B^@^AA^@T^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^AM-P^P^C^P^B^@^_E^@^AT^KM-V^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^BM-Q^P^C^P^B^@^_E^@^B^@^@x^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^CM-R^P^C^P^B^@^_E^@^CM-h&M-^G^P^C^P^BxD^@M-h&M-\^P^C^P^B^@^_D^@M-h&^@M-^C^P^C^P^BxAM-K^P^C^P^B^@^AA^@T^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^DM-S^P^C^P^B^@^_E^@^DM-<^D:^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^EM-T^P^C^P^B^@^_E^@^EW^AM-S^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^FM-U^P^C^P^B^@^_E^@^F^@PM-L^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^GM-V^P^C^P^B^@^_E^@^GM-^@^S^P^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxD^@M-H^^M-4^P^C^P^B^@^_D^@M-H^^^@[^P^C^P^BxAM-K^P^C^P^B^@^AA^@T^P^C^P^BxE^@^HM-W^P^C^P^B^@^_E^@^H^K^@M-^I^P^C^P^BxD^@M-H^^M-4^P^C^P^B^@^_D^@M-H^^^@[^P^C^P^BxAM-K^P^C^P^B^@^AA^@T^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^IM-X^P^C^P^B^@^_E^@^I^@^@^?^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@$
M-Y^P^C^P^B^@^_E^@$
M-n^Ao^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^KM-Z^P^C^P^B^@^_E^@^KM-^?M-^?^?^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^A^FM-V^P^C^P^B^@^_E^A^FM-^?M-^?{^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxBM-L^P^C^P^B^@^AB^@U^P^C^P^BxBM-L^P^C^P^B^@^AB^@U^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^AM-P^P^C^P^B^@^_E^@^A|^P^C^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^BM-Q^P^C^P^B^@^_E^@^B^@^@x^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^CM-R^P^C^P^B^@^_E^@^CM-H^^_^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^DM-S^P^C^P^B^@^_E^@^Dp^BM-l^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^EM-T^P^C^P^B^@^_E^@^EM-!^@^\^P^C^P^BxBM-L^P^C^P^B^@^AB^@U^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^FM-U^P^C^P^B^@^_E^@^FM-^@PL^P^C^P^BxD^@M-x*M-p^P^C^P^B^@^_D^@M-x*^@M-^W^P^C^P^BxAM-K^P^C^P^B^@^AA^@T^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^GM-V^P^C^P^B^@^_E^@^GM-^@^S^P^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^HM-W^P^C^P^B^@^_E^@^H^K^@M-^I^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^IM-X^P^C^P^B^@^_E^@^I^@^@^?^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@$
M-Y^P^C^P^B^@^_E^@$
i^@M-i^P^C^P^BxD^@M-x*M-p^P^C^P^B^@^_D^@M-x*^@M-^W^P^C^P^BxAM-K^P^C^P^B^@^AA^@T^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^KM-Z^P^C^P^B^@^_E^@^KM-^?M-^?^?^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^A^FM-V^P^C^P^B^@^_E^A^FM-^?M-^?{^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^@M-O^P^C^P^B^@^_E^@^@<^VM-H^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^AM-P^P^C^P^B^@^_E^@^A6^KM-8^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^BM-Q^P^C^P^B^@^_E^@^B^@^@x^P^C^P^BxD^@M-x*M-p^P^C^P^B^@^_D^@M-x*^@M-^W^P^C^P^BxAM-K^P^C^P^B^@^AA^@T^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^CM-R^P^C^P^B^@^_E^@^CM-x*M-^[^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^DM-S^P^C^P^B^@^_E^@^DM-0^D.^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^EM-T^P^C^P^B^@^_E^@^EL^AM-H^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^FM-U^P^C^P^B^@^_E^@^F@P^L^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^GM-V^P^C^P^B^@^_E^@^G^@^SM-^P^P^C^P^BxD^@M-x*M-p^P^C^P^B^@^_D^@M-x*^@M-^W^P^C^P^BxAM-K^P^C^P^B^@^AA^@T^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^HM-W^P^C^P^B^@^_E^@^H^K^@M-^I^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@^IM-X^P^C^P^B^@^_E^@^I^@^@^?^P^C^P^BxCM-M^P^C^P^B^@^_C^K^@^@^@^?^P^C^P^BxE^@$
M-Y^P^C^P^B^@^_E^@$
^Q^AM-^R^P^C^P^BxCM-M^P^C^C
`

Out of interest, why cant the Pi control the motor directly without the control panel?  It seems like the motors board itself is doing modbus etc? (Silly question perhaps).

Let me know how I can do any work to assist here, even if i have to play around

**Reply by sfeakes** [reply]
This explains why AqualinkD can’t control the pump directly.  Basically the way the protocol works is there can only be one master, that master has to control all other devices on the RS485 bus and the control panel is fixed at being the master.
https://github.com/sfeakes/AqualinkD/discussions/365#discussioncomment-11364230
If you put the pump on a separate (it’s own) RS485 bus, then you can control it from a raspberryPI, but AqualinkD is not designed to do that. 

I’m a bit confused as to exactly what are you actually trying to achieve here though.

- If it’s control the pump directly and not through a control panel, then there are other projects that are aimed at doing exactly that.  (Like this https://github.com/tagyoureit/nodejs-poolController ). They may not support your pump though.

- If it’s to get AqualinkD to work, then you should be focusing in getting AqualinkD communicating with the control panel and not the pump.

BTW. the term “control panel”,  means Jandy’s RS / PDA control panel (something like this https://www.jandy.com/en/products/pool-automation/automation-system/aqualink-rs-automation).  Not interface on the pump.

**Comment by odgeza** [comment]
Ok, that makes a few pieces fall into place in my puzzle. 

I like the idea that the controller is more reliable than my Pi! So I think controlling the speed and on of status os all I'm really looking to do. I have a solar heater, and I want to 1: vary the pump through the day a few times while the solar heater is enabled, 2: run the pump a bit faster (using more power) when it's high summer in the middle of the day. 3: just vary the speed of the pump in general, the cleaned seems to like some spots more at certain rpm

**Comment by odgeza** [comment]
Extra note: I would probably replace my salt cholirnator at some point with something a bit more fancy when it's electrode needs replacing. It doesn't. I would also like to add a heat pump/thermometer over time, and keep it all linked up.
